### PR TITLE
feat: Message Parser for wholesale settlement json requests

### PR DESCRIPTION
**Pull request by MadsDue**
## Description
We are now validates the request against the json schema and parsing the request to a type within the incoming message module.

### CODE REVIEW

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+using Energinet.DataHub.EDI.IncomingMessages.Infrastructure.ValidationErrors;
+using Energinet.DataHub.EDI.IncomingMessages.Interfaces;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Energinet.DataHub.EDI.Tests.CimMessageAdapter.Messages.WholesaleSettlementMessageParsers;
+
+public class MessageParserTests
+{
+    private const string SeriesDummy = "{\"mRID\":\"25836143\",\"aggregationSeries_Period.resolution\":\"PT1M\",\"chargeTypeOwner_MarketParticipant.mRID\":{\"codingScheme\":\"A10\",\"value\":\"570001110111\"},\"end_DateAndOrTime.dateTime\": \"2022-08-31T22:00:00Z\",\"energySupplier_MarketParticipant.mRID\":{\"codingScheme\":\"A10\",\"value\":\"5799999933318\"},\"meteringGridArea_Domain.mRID\":{\"codingScheme\": \"NDK\",\"value\":\"244\"},\"settlement_Series.version\":{\"value\":\"D01\"},\"start_DateAndOrTime.dateTime\":\"2022-08-17T22:00:00Z\",\"ChargeType\":[{\"mRID\":\"EA-001\",\"type\":{\"value\":\"D03\"}}]},";
```
Comment (MWO1024) [inline source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:35]: Maybe we could use a verbatim string? I will take up multiple lines, but might be easier for a human to read?

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    [Theory]
+    [MemberData(nameof(CreateMessagesWithSingleAndMultipleTransactions))]
+    public async Task Successfully_parsed(DocumentFormat format, Stream message)
+    {
+        var result = await _marketMessageParser.ParseAsync(new IncomingMessageStream(message), format, IncomingDocumentType.RequestWholesaleSettlement, CancellationToken.None);
+        using var assertionScope = new AssertionScope();
+        Assert.True(result.Success);
+        var marketMessage = (WholesaleSettlementMessage)result!.IncomingMessage!;
+        Assert.True(marketMessage != null);
```
Comment (MWO1024) [inline source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:80]: `marketMessage.Should().NotBeNull()`?

File: source/IncomingMessages.Application/Extensions/DependencyInjection/IncomingMessagesExtensions.cs [hunk 3]
```diff
@@ -57,6 +58,7 @@ public static IServiceCollection AddIncomingMessagesModule(this IServiceCollecti
             .AddScoped<IMessageParser, XmlMessageParser>()
             .AddScoped<IMessageParser, JsonMessageParser>()
             .AddScoped<IMessageParser, B2CJsonMessageParser>()
+            .AddScoped<IMessageParser, WholesaleSettlementMessageParser>()
```
Comment (RasmusGraabaek) [inline source/IncomingMessages.Application/Extensions/DependencyInjection/IncomingMessagesExtensions.cs:61]: Should we prefix the aggregated... message parsers?

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        Assert.Equal("12345678", marketMessage.MessageId);
+        Assert.Equal("D05", marketMessage.BusinessReason);
+        Assert.Equal("5799999933318", marketMessage.SenderNumber);
+        Assert.Equal("DDQ", marketMessage.SenderRoleCode);
+        Assert.Equal("5790001330552", marketMessage.ReceiverNumber);
+        Assert.Equal("DDZ", marketMessage.ReceiverRoleCode);
+        Assert.Equal("2022-12-17T09:30:47Z", marketMessage.CreatedAt);
+        Assert.Equal("23", marketMessage.BusinessType);
+
+        foreach (var serie in marketMessage.Series)
```
Comment (MWO1024) [inline source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:90]: Could we rename to `series` as `serie` is in fact not a proper word. Series is one of those annoying uncountable words in English 😭

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    }
+
+    [Theory]
+    [MemberData(nameof(CreateBadMessages))]
+    public async Task Messages_with_errors(DocumentFormat format, Stream message, string expectedError)
+    {
+        var result = await _marketMessageParser.ParseAsync(new IncomingMessageStream(message), format, IncomingDocumentType.RequestWholesaleSettlement, CancellationToken.None);
+
+        Assert.True(result.Errors.Count > 0);
```
Comment (MWO1024) [inline source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:116]: `result.Should().NotBeEmpty()`?

Comment (MWO1024) [reply source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:116]: And something similar for the other `Assert.True` statements

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        Assert.Equal("5799999933318", marketMessage.SenderNumber);
+        Assert.Equal("DDQ", marketMessage.SenderRoleCode);
+        Assert.Equal("5790001330552", marketMessage.ReceiverNumber);
+        Assert.Equal("DDZ", marketMessage.ReceiverRoleCode);
+        Assert.Equal("2022-12-17T09:30:47Z", marketMessage.CreatedAt);
+        Assert.Equal("23", marketMessage.BusinessType);
+
+        foreach (var serie in marketMessage.Series)
+        {
+            Assert.True(serie != null);
```
[the remark this answers is not in the document]
Comment (MWO1024) [reply source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:92]: The naming thing here too

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs (hunk 1 above)
Comment (MadsDue) [reply source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:35]: I am not sure if it is necessary to understand/read the element :/ that is also why it is called "Dummy".

**Review by RasmusGraabaek — DISMISSED** [review]

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs (hunk 2 above)
Comment (MadsDue) [reply source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:80]: We haven't used fluentvalidation here yet. I will take it in another PR 👍

File: source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs (hunk 4 above)
Comment (MadsDue) [reply source/Tests/CimMessageAdapter/Messages/WholesaleSettlementMessageParsers/MessageParserTests.cs:90]: Good catch 👍  I will add it to my next pr.

**Review by MWO1024 — APPROVED** [review]

**Pull request merged**